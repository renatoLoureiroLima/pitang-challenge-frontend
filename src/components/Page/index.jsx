import { Container } from 'react-bootstrap';

import Card from '../Card';

function index({ title, children }) {
  return (
    <Container className='mt-4' data-testid='page-test'>
      <Card title={title}>{children}</Card>
    </Container>
  );
}

export default index;
