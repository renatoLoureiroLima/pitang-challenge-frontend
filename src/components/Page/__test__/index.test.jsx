import ReactDOM from 'react-dom';
import Page from '../index';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Page></Page>, div);
});

it('renders page correctly', () => {
  const { getByTestId } = render(<Page title='Test'></Page>);
  expect(getByTestId('page-test')).toHaveTextContent('Test');
});

it('matches snapshot', () => {
  const tree = renderer.create(<Page title='Test'></Page>).toJSON();
  expect(tree).toMatchSnapshot();
});
