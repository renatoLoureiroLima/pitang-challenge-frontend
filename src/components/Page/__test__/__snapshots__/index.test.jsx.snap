// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="mt-4 container"
  data-testid="page-test"
>
  <div
    className="card"
    data-testid="card-test"
  >
    <div
      className="card-header"
    >
      <div
        className="card-title h5"
      >
        Test
      </div>
    </div>
    <div
      className="card-body"
    />
  </div>
</div>
`;
